cmd/state-svc/internal/resolver/analytics/analytics.go

```diff
@@ -119,41 +119,15 @@ func NewResolver(cfg *config.Instance) *Resolver {
 // AnalyticsEvent schedules the sending of the analytics event
 func (r *Resolver) AnalyticsEvent(_ context.Context, category, action string, label, projectName, out, userID *string) (*graph.AnalyticsEventResponse, error) {
 	logging.Debug("Analytics event resolver")
-
-	lbl := ""
-	if label != nil {
-		lbl = *label
-	}
-
-	pn := ""
-	if projectName != nil {
-		pn = *projectName
-	}
-
-	o := string(output.PlainFormatName)
-	if out != nil {
-		o = *out
-	}
-
-	uid := ""
-	if userID != nil {
-		uid = *userID
-	}
+	ev := event.New(category, action, label, projectName, out, userID)
 
 	// We do not wait for the events to be processed, just scheduling them
 	go func() {
 		select {
-		case r.events <- deferred.EventData{
-			Category:    category,
-			Action:      action,
-			Label:       lbl,
-			ProjectName: pn,
-			Output:      o,
-			UserID:      uid,
-		}:
+		case r.events <- ev:
 		case <-r.ctx.Done():
 			// try to defer event if it cannot be scheduled in this session
-			_ = deferred.DeferEvent(category, action, lbl, pn, o, uid)
+			_ = deferred.DeferEvent(ev)
```

Comment:
I don't think the state-svc should "defer" anything. We need to get rid of the concept of deferring altogether. State-svc is the solution. If the state-svc can't send an event then we need log an error so we can address it.

ie. the situation in which it cannot send an event is an error and should not be worked around.